src/callbacks_stage/subcell_bounds_check_2d.jl

```diff
@@ -12,35 +12,25 @@
     (; variable_bounds) = limiter.cache.subcell_limiter_coefficients
     (; idp_bounds_delta_local, idp_bounds_delta_global) = limiter.cache
 
-    # Note: Accessing the threaded memory vector `idp_bounds_delta_local` with
-    # `deviation = idp_bounds_delta_local[key][Threads.threadid()]` causes critical performance
-    # issues due to False Sharing.
-    # Initializing a vector with n times the length and using every n-th entry fixes this
-    # problem and allows proper scaling:
-    # `deviation = idp_bounds_delta_local[key][n * Threads.threadid()]`
-    # Since there are no processors with caches over 128B, we use `n = 128B / size(uEltype)`
-    stride_size = div(128, sizeof(eltype(u))) # = n
-
     if local_minmax
         for v in limiter.local_minmax_variables_cons
             v_string = string(v)
             key_min = Symbol(v_string, "_min")
             key_max = Symbol(v_string, "_max")
-            deviation_min_threaded = idp_bounds_delta_local[key_min]
-            deviation_max_threaded = idp_bounds_delta_local[key_max]
-            @threaded for element in eachelement(solver, cache)
-                deviation_min = deviation_min_threaded[stride_size * Threads.threadid()]
-                deviation_max = deviation_max_threaded[stride_size * Threads.threadid()]
+            deviation_min = idp_bounds_delta_local[key_min]
+            deviation_max = idp_bounds_delta_local[key_max]
+            @batch reduction=((max, deviation_min), (max, deviation_max)) for element in eachelement(solver,
```

Comment:
```suggestion
            @threaded reduction=((max, deviation_min), (max, deviation_max)) for element in eachelement(solver,
```
In the current implementation, `@threaded` is basically just an alias for `Polyester.@batch` (see the [implementation](https://github.com/trixi-framework/Trixi.jl/blob/main/src/auxiliary/auxiliary.jl#L219C1-L250C4)). So for consistency I think it would be nice to use `@threaded`. However, I think the `reduction` keyword is not supported by `Threads.@threads`, right? This means, we cannot easily switch back from `Polyester.@batch` to `Threads.@threads` because this would not run anymore. Any opinions?

Reply:
I thought about that as well. But as you said, the idea of using `@threaded` and not directly `Trixi.@batch` is to possibly change if everywhere effortless, what's not possible here. That's why I kept it this way. But if anyone has a better idea or another opinion, I'll gladly change it.

Reply:
Yes, I'm fine to keep `@batch`. I just wanted to hear some other opinions.

Reply:
Fine with me as well. Please add a comment explaining this here.